src/beautifiers/markdown-blocks/index.coffee

```diff
@@ -0,0 +1,52 @@
+module.exports = (text, logger) ->
+  return new @Promise((resolve, reject) ->
+    extractCodeBlocks = require 'gfm-code-blocks'
+    beautifier = require '../../beautifier'
+    cleanMarkdown = text
+    codeBlocks = extractCodeBlocks(cleanMarkdown)
+    beautifyBlockPromises = []
+    for codeBlock in codeBlocks
+      codeLanguage = codeBlock.lang.match(/^[\w\d]+/)?[0]
+      logger.verbose "Beautify code block #{codeLanguage}"
+      if !codeLanguage
+        beautifyBlockPromises.push(null)
+      else
+        beautifyBlockPromises.push(new @Promise((resolve, reject) ->
```

Comment:
It appears all of the branches lead to a `beautifyBlockPromises.push`.

I think using `Array.map` would be better for this. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map

Reply:
Good point! I was not pushing nulls in the very first implementation, so did not consider Array.map as an option here.